climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,891 @@
+        The value per point is value_per_pnt
+
+    """
+
+    gdf_agg = gdf_pnts.copy()
+    gdf_agg['value'] = value_per_pnt
+
+    return gdf_agg
+
+
+def poly_to_pnts(gdf, res):
+    """
+    Disaggragate (multi-)polygons geodataframe to points
+
+    Parameters
+    ----------
+    gdf : geodataframe
+        Can be any CRS
+    res : float
+        Resolution (same units as gdf crs)
+
+    Returns
+    -------
+    geodataframe
+        Geodataframe with a double index, first for polygon geometries,
+        second for the point disaggregation of the polygons.
+
+    """
+
+    # Needed because gdf.explode() requires numeric index
+    idx = gdf.index.to_list() #To restore the naming of the index
+
+    gdf_points = gdf.copy().reset_index(drop=True)
+    gdf_points['geometry_pnt'] = gdf_points.apply(
+        lambda row: _interp_one_poly(row.geometry, res), axis=1)
+    gdf_points = _swap_geom_cols(
+        gdf_points, geom_to='geometry_orig', new_geom='geometry_pnt'
+        )
+
+    gdf_points = gdf_points.explode()
+    gdf_points.index = gdf_points.index.set_levels(idx, level=0)
+    return gdf_points
+
+def poly_to_pnts_m(gdf, res):
+    """
+    Disaggragate (multi-)polygons geodataframe to points
+
+    Parameters
+    ----------
+    gdf : geodataframe
+        Can be any CRS
+    res : float
+        Resolution in meters
+
+    Returns
+    -------
+    geodataframe
+        Geodataframe with a double index, first for polygon geometries,
+        second for the point disaggregation of the polygons.
+
+    """
+
+    orig_crs = gdf.crs
+
+    # Needed because gdf.explode() requires numeric index
+    idx = gdf.index.to_list() # To restore the naming of the index.
+
+    gdf_points = gdf.copy().reset_index(drop=True)
+    gdf_points['geometry_pnt'] = gdf_points.apply(
+        lambda row: _interp_one_poly_m(row.geometry, res, orig_crs), axis=1)
+    gdf_points = _swap_geom_cols(
+        gdf_points, geom_to='geometry_orig', new_geom='geometry_pnt'
+        )
+    gdf_points = gdf_points.explode()
+    gdf_points.index = gdf_points.index.set_levels(idx, level=0)
+    return gdf_points
+
+def _interp_one_poly(poly, res):
+    """
+    Disaggragate a single polygon to points
+
+    Parameters
+    ----------
+    poly : shapely Polygon
+        Polygon
+    res : float
+        Resolution (same units as gdf crs)
+
+    Returns
+    -------
+    shapely multipoint
+        Grid of points rasterizing the polygon
+
+    """
+
+    if poly.is_empty:
+        return shgeom.MultiPoint([])
+
+    height, width, trafo = u_coord.pts_to_raster_meta(poly.bounds, (res, res))
+    x_grid, y_grid = u_coord.raster_to_meshgrid(trafo, width, height)
+    in_geom = sh.vectorized.contains(poly, x_grid, y_grid)
+
+    if sum(in_geom.flatten()) > 1:
+        return shgeom.MultiPoint([(x, y) for x, y in
+                            zip(x_grid[in_geom], y_grid[in_geom])])
+    else:
+        return shgeom.MultiPoint([poly.representative_point()])
```

Comment:
should this return also a warning/info? because it assigns a point even though it's not representative of the usual expectable area per point. Esp. for value assignment / disaggregation later with a fixed value per point (assuming a certain underlying area), this could lead to distortions if this happens very often

Reply:
Maybe... it would be hard though to interpret for the user right? The open method has a warning in the docstring about that.

Reply:
mmmh, how / where? didn't find it.

Reply:
It looks like it got lost over the iterations. You are right, one should add it in the docstring of the open method.